includes/classes/class-plugin-manager.php

```diff
@@ -0,0 +1,409 @@
+							);
+						}
+					}
+					break;
+
+				case 'latest_installed':
+				case 'newer_installed':
+					if ( is_plugin_active( $status['file'] ) ) {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							_x( 'Active', 'plugin', 'default' )
+						);
+						if ( current_user_can( 'deactivate_plugin', $status['file'] ) ) {
+							global $page;
+							$s       = isset( $_REQUEST['s'] ) ? $_REQUEST['s'] : ''; // phpcs:ignore
+							$context = $status['status'];
+
+							$action_links[] = sprintf(
+								'<a href="%s" id="deactivate-%s" aria-label="%s" style="color:red;text-decoration: underline;">%s</a>',
+								wp_nonce_url( 'plugins.php?wpp=1&action=deactivate&amp;plugin=' . rawurlencode( $status['file'] ) . '&amp;plugin_status=' . $context . '&amp;paged=' . $page . '&amp;s=' . $s, 'deactivate-plugin_' . $status['file'] ),
+								esc_attr( $plugin['slug'] ),
+								/* translators: %s: Plugin name. */
+								esc_attr( sprintf( _x( 'Deactivate %s', 'plugin', 'default' ), $plugin['slug'] ) ),
+								__( 'Deactivate', 'default' )
+							);
+						}
+					} elseif ( current_user_can( 'activate_plugin', $status['file'] ) ) {
+						if ( $compatible_php && $compatible_wp ) {
+							$button_text = __( 'Activate', 'default' );
+							/* translators: %s: Plugin name. */
+							$button_label = _x( 'Activate %s', 'plugin', 'default' );
+							$activate_url = add_query_arg(
+								array(
+									'_wpnonce' => wp_create_nonce( 'activate-plugin_' . $status['file'] ),
+									'action'   => 'activate',
+									'plugin'   => $status['file'],
+									'wpp'      => 1,
+								),
+								network_admin_url( 'plugins.php' )
+							);
+
+							if ( is_network_admin() ) {
+								$button_text = __( 'Network Activate', 'default' );
+								/* translators: %s: Plugin name. */
+								$button_label = _x( 'Network Activate %s', 'plugin', 'default' );
+								$activate_url = add_query_arg( array( 'networkwide' => 1 ), $activate_url );
+							}
+
+							$action_links[] = sprintf(
+								'<a href="%1$s" class="button activate-now" aria-label="%2$s">%3$s</a>',
+								esc_url( $activate_url ),
+								esc_attr( sprintf( $button_label, $plugin['name'] ) ),
+								$button_text
+							);
+						} else {
+							$action_links[] = sprintf(
+								'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+								_x( 'Cannot Activate', 'plugin', 'default' )
+							);
+						}
+					} else {
+						$action_links[] = sprintf(
+							'<button type="button" class="button button-disabled" disabled="disabled">%s</button>',
+							_x( 'Installed', 'plugin', 'default' )
+						);
+					}
+					break;
+			}
+		}
+
+		$details_link = self_admin_url(
+			'plugin-install.php?tab=plugin-information&amp;plugin=' . $plugin['slug'] .
+			'&amp;TB_iframe=true&amp;width=600&amp;height=550'
```

Comment:
Why include the `&amp;` in the URL here? This should rather be handled by `esc_url()`. Probably the same as https://github.com/WordPress/performance/pull/864/files#r1367167618

Reply:
@westonruter I borrowed heavily from WP core's implementation of `WP_Plugin_Install_List_Table` which uses this approach in all of the admin URLs it generates, see https://github.com/WordPress/WordPress/blob/master/wp-admin/includes/class-wp-plugin-install-list-table.php#L655. Although, most implementations just make use of a single action or similar, so this may have been an oversight in core.

Regardless, your feedback is valid and I will apply this suggestion 👍

Reply:
@westonruter I've addressed this in https://github.com/WordPress/performance/pull/864/commits/4ba92e3c4bf551eabfd83ba899ca044462f2a088, please let me know if you're happy with said approach, I opted for `admin_url` wrapped in `add_query_arg` and `esc_url_raw`.